### PR TITLE
Feature/user accounts

### PR DESCRIPTION
**Pull request by christophmeissner**
TL;DR: 
* removes the custom source code of registration package and replaces it with pip package 'django-registration-redux=1.2'

Also:
* removes unused ValueStore model
* introduces a new model accounts.models.UserAccount, which is created, once an user has completed the signup process by confirming his email address
* introduces a new model scheduler.models.ShiftHelper, which is the m2m-through model for Need.helpers relating users to needs when signing up for shifts
* refactored and moved code for finding conflicting shifts to scheduler.models.ShiftHelperManager.conflicts(need, user_account=None, grace=None)
* some cosmetics and added some flavours

### CODE REVIEW

**Comment by pitpalme** [comment]
LGTM ...

File: accounts/models.py [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+                                related_name='account')
+
+    class Meta:
+        verbose_name = _('user account')
+        verbose_name_plural = _('user accounts')
+
+@receiver(user_activated)
+def registration_completed(sender, user, request, **kwargs):
+    account, created = UserAccount.objects.get_or_create(user=user)
+    print account, created
```
Comment (maikhoepfel) [inline accounts/models.py]: Leftover `print` statement

File: scheduler/views.py [hunk 2]
```diff
@@ -75,36 +72,52 @@ def form_invalid(self, form):
...
+        except UserAccount.DoesNotExist:
+            messages.warning(self.request, _(u'User account does not exist.'))
             return super(PlannerView, self).form_valid(form)
 
-        join_shift = form.cleaned_data.get("join_shift")
-        leave_shift = form.cleaned_data.get("leave_shift")
+        shift_to_join = form.cleaned_data.get("join_shift")
+        shift_to_leave = form.cleaned_data.get("leave_shift")
 
         print(form.cleaned_data)
```
Comment (maikhoepfel) [inline scheduler/views.py]: I'll see about cleaning them.

File: tests/scheduler/test_models.py [hunk 3]
```diff
@@ -22,24 +28,89 @@ class NeedTestCase(TestCase):
...
+    def test_conflict_at_end_no_grace(self):
+        need = create_need(11, 15)
+        assert ShiftHelper.objects.conflicting(need=need,
+                                               grace=None).count() == 1
+
+    def test_conflict_within(self):
+        need = create_need(10, 11)
+        assert ShiftHelper.objects.conflicting(need=need).count() == 1
+
+    def test_conflict_within_no_grace(self):
```
Comment (maikhoepfel) [inline tests/scheduler/test_models.py]: Yay for tests!

**Comment by maikhoepfel** [comment]
Lovely work, thanks for doing this. I think we need to start documenting what our individual apps are used for; I'm not sure I understand the purpose of the `registration_history` app.
